wherehows-web/app/routes/search.ts

```diff
@@ -26,9 +34,21 @@ export default class SearchRoute extends Route.extend(AuthenticatedRouteMixin) {
   @action
   loading(transition: import('ember').Ember.Transition): void {
     let controller = this.controllerFor('search');
-    controller.set('searchLoading', true);
+    set(controller, 'searchLoading', true);
     transition.promise!.finally(function() {
-      controller.set('searchLoading', false);
+      set(controller, 'searchLoading', false);
     });
   }
+
+  /**
+   * In order to keep the service up date with the state. The router pass
+   * the keyword from the queryParams to the service.
+   * @param transition Ember transition
+   */
+  @action
+  willTransition(transition: import('ember').Ember.Transition & { targetName: string }) {
```

Comment:
Should we just define Transition in our own .d.ts file instead of using `import('ember').Ember.Transition` everytime? Seems like out of place syntax to me.

Reply:
I was trying to follow same pattern. @theseyi thoughts?